### PR TITLE
Code revision

### PR DESCRIPTION
**Pull request by gabzn**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,4 +17,9 @@
...
     git add .
     git commit -m "comments" 
-    git push origin <branch-name>
\ No newline at end of file
+    git push origin <branch-name>
+
+## merge main into the current branch
+    git checkout <branch-name>  (stay in the branch or switch to the branch)
+    git fetch
+    git merge origin main       (merge everything in main into the current branch)
```
Comment (gabzn) [inline README.md:25]: Added instructions on how to merge main into the current branch.

File: server/routes/authRoute.js [hunk 2]
```diff
@@ -22,10 +22,9 @@ router.get('/', auth, async (req, res) => {
 	}
 });
 
-// @route   GET app/users
-// @desc    Authenticate user & get token
+// @route   POST app/auth
+// @desc    Login user & Return JWT token
```
Comment (gabzn) [inline server/routes/authRoute.js:26]: Changed wordings.

File: server/routes/authRoute.js [hunk 3]
```diff
@@ -45,15 +44,15 @@ router.post(
...
+					.json({ errors: 'Invalid Credentials'});
 			}
 			/* Check if the user matches the password */
 			const isMatch = await bcrypt.compare(password, user.password);
 
 			if (!isMatch) {
 				return res
 					.status(400)
-					.json({ errors: [{ msg: 'Invalid Credentials' }] });
+					.json({ errors: 'Invalid Credentials'});
```
Comment (gabzn) [inline server/routes/authRoute.js:55]: Instead of returning an array, changed to return an object.

File: server/routes/billRoute.js [hunk 4]
```diff
@@ -78,11 +71,31 @@ router.post(
...
+		const targetDish = await Bill.findOne(
+			{_id: billID},
+			{dishes: {$elemMatch: {_id: dishID}}}
+		);
+		targetDish.dishes[0].dishName = dishName;
+		await targetDish.save();
+
+		res.json({message: 'Dish has been updated,'});
+	}
+)
```
Comment (gabzn) [inline server/routes/billRoute.js:99]: Added route to update a certain dish.

File: server/server.js [hunk 5]
```diff
@@ -16,20 +15,20 @@ app.get('/', (req, res, next) => {
...
+// 	let imageInBase64 = req.body.file;
+// 	imageInBase64 = imageInBase64.split(',');
+// 	imageInBase64 = imageInBase64[1];
 
-	await parseImage(imageInBase64);
-	res.json({ message: 'Upload was successful' });
-});
+// 	await parseImage(imageInBase64);
+// 	res.json({ message: 'Upload was successful' });
+// });
```
Comment (gabzn) [inline server/server.js:26]: Commented out the route for API since no longer needed.

File: server/server.js [hunk 6]
```diff
@@ -16,20 +15,20 @@ app.get('/', (req, res, next) => {
...
 
 app.use('/app/dashboard', require('./routes/dashboardRoute'));
-app.use('/app/users', require('./routes/userRoute'));
+app.use('/app/register', require('./routes/registerRoute'));
 app.use('/app/auth', require('./routes/authRoute'));
-app.use('/app/bill', billRoute);
+app.use('/app/bill', require('./routes/billRoute'));
```
Comment (gabzn) [inline server/server.js:31]: Changed for consistency.

File: server/routes/billRoute.js [hunk 7]
```diff
@@ -6,7 +6,7 @@ const auth = require('../../middleware/auth');
 const { check, validationResult } = require('express-validator');
 // const Dish = require('../models/dish');
 
-// @route   POST app/bill
+// @route   POST app/bill/
```
Comment (WenxuC) [inline server/routes/billRoute.js:9]: is there a reason for the extra /?

File: server/routes/billRoute.js (hunk 4 above)
Comment (WenxuC) [reply server/routes/billRoute.js:99]: could you wrap this in a try catch block similar to the ones I have in authRoute, make sure all the routes have one.

**Review by WenxuC — CHANGES_REQUESTED** [review]
Add try catch blocks to routes

File: server/routes/billRoute.js (hunk 4 above)
Comment (gabzn) [reply server/routes/billRoute.js:103]: Fixed.

File: server/routes/billRoute.js [hunk 8]
```diff
@@ -87,14 +87,18 @@ router.post(
...
 
-		const targetDish = await Bill.findOne(
-			{_id: billID},
-			{dishes: {$elemMatch: {_id: dishID}}}
-		);
-		targetDish.dishes[0].dishName = dishName;
-		await targetDish.save();
-
-		res.json({message: 'Dish has been updated,'});
+		try {
```
Comment (gabzn) [inline server/routes/billRoute.js:90]: Added try catch block.

File: server/routes/billRoute.js (hunk 7 above)
Comment (gabzn) [reply server/routes/billRoute.js:9]: The extra / does not affect anything.

**Pull request merged**